Modules/unicodedata.c

```diff
@@ -1253,6 +1274,112 @@ unicodedata_UCD_lookup_impl(PyObject *self, const char *name,
     return PyUnicode_FromOrdinal(code);
 }
 
+typedef struct {
+    PyObject_HEAD
+    PyObject* str;
+    int pos;
+} GraphemeClusterIterator;
+
+void GCI_Del(PyObject* x)
+{
+    GraphemeClusterIterator *i = (GraphemeClusterIterator *)x;
+    Py_DECREF(i->str);
+    PyObject_Del(x);
+}
+
+static PyObject* GCI_iter(PyObject *self)
+{
+    Py_INCREF(self);
+    return self;
+}
+
+#include "grapheme_cluster_break_automaton.h"
+
+PyObject* GCI_iternext(PyObject *self)
+{
+    GraphemeClusterIterator *p = (GraphemeClusterIterator *)self;
+    int kind = PyUnicode_KIND(p->str);
+    void *pstr = PyUnicode_DATA(p->str);
+    if (PyUnicode_READ(kind, pstr, p->pos)) {
+        int start = p->pos;
+        GCBState s = STATE_sot;
+        while (1) {
+            if (!PyUnicode_READ(kind, pstr, p->pos)) {
+                return PyUnicode_Substring(p->str, start, p->pos);
+            }
+            Py_UCS4 chr = PyUnicode_READ(kind, pstr, p->pos);
+            int prop = _getrecord_ex(chr)->grapheme_cluster_break;
+            s = GRAPH_CLUSTER_AUTOMATON[s][prop];
+            if (s == STATE_BREAK) {
+                return PyUnicode_Substring(p->str, start, p->pos);
+            }
+            ++p->pos;
+        }
+    } else {
+        /* Raising of standard StopIteration exception with empty value. */
+        PyErr_SetNone(PyExc_StopIteration);
+        return NULL;
+    }
+}
+
+static PyTypeObject GraphemeClusterIteratorType = {
+    PyVarObject_HEAD_INIT(NULL, 0)
+    "unicodedata.GraphemeClusterIterator", /*tp_name*/
+    sizeof(GraphemeClusterIterator),       /*tp_basicsize*/
+    0,                         /*tp_itemsize*/
+    (destructor)GCI_Del,       /*tp_dealloc*/
+    0,                         /*tp_print*/
```

Comment:
Please use designated initializers for new types.

```
...
.tp_dealloc = (destructor)GCI_Del,
.tp_flags = Py_TPFLAGS_DEFAULT,
.tp_doc = "Internal grapheme cluster iterator object",
.tp_iter = GCI_iter,
.tp_iternext = GCI_iternext
};
```

Reply:
Fixed